src/components/MagicCodeInput.js

```diff
@@ -103,11 +109,28 @@ function MagicCodeInput(props) {
         },
     }));
 
+    /**
+     * Check if current magic code is ready for submit
+     * @returns {boolean}
+     */
+    const isReadyToSubmit = () => {
+        const numbers = decomposeString(props.value);
+        return props.shouldSubmitOnComplete && _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length === CONST.MAGIC_CODE_LENGTH && !props.network.isOffline;
+    };
+
+    useOnNetworkReconnect(() => {
+        if (!isReadyToSubmit()) {
+            return;
+        }
+        inputRefs.current[editIndex].blur();
+        setFocusedIndex(undefined);
+        props.onFulfill(props.value);
```

Comment:
The whole is redundant I mentioned it [here](https://github.com/Expensify/App/issues/18311#issuecomment-1538541792) already. Check lines 133 to 138 can we this into a function. `isReadyToSubmit` is not need can be moved in the new method. Ex: `validateAndSubmit`

Reply:
That's fast 🚀